src/api/ppx_deriving.cppo.ml

```diff
@@ -762,5 +807,5 @@ let hash_variant s =
    driver. *)
 let () =
   Migrate_parsetree.Driver.register ~name:"ppx_deriving"
-    (module Migrate_parsetree.OCaml_current)
+    (module Migrate_parsetree.OCaml_408)
     (fun _ _ -> mapper)
```

Comment:
Note: `api/ppx_deriving` here registers the mapper into the o-m-p driver (for use with other o-m-p-enabled transformations). In `src/ppx_deriving_main` there is another registration mechanism, using `Ast_mapper` from the compilerlibs, which is only used when producing an executable-style preprocessor (instead of being linked as a plugin of a driver executable). The two registration mechanisms seem to cohabit nicely enough, and in any case were already present before this PR.

Reply:
Would it be possible to use ppxlib's registration mechanism? Even if you don't use the context free api, it still has some advantages.